### PR TITLE
[AKS] update to v20180331 API and add new feature args

### PR DESCRIPTION
**Pull request by mboersma**
Uses the new [v20180331 AKS API](https://github.com/Azure/azure-rest-api-specs/pull/3131) and adds arguments to support new AKS GA and preview features: RBAC, VNet customization ("advanced networking"), no SSH key needed, Log Analytics monitoring, HTTP application routing, and Azure AD integration (in preview).

Note that this still includes a temp commit with a private copy of the azure-mgmt-containerservice .whl. I'll need to remove that when it is published to Pypi, before this PR merges.

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by troydai** [comment]
@mboersma I will put up a do not merge label per the private build. When will the private package be online?

**Comment by mboersma** [comment]
> When will the private package be online?

I'm not sure yet. Azure/azure-sdk-for-python#2186 is ready to go and I'm asking around to get it published.

File: src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py [hunk 1]
```diff
@@ -1350,13 +1373,56 @@ def aks_create(cmd, client, resource_group_name, name, ssh_key_value,  # pylint:
...
+            dns_service_ip=dns_service_ip,
+            docker_bridge_cidr=docker_bridge_address
+        )
+
+    addon_profiles = {}
+    addons = enable_addons.split(',') if enable_addons else []
+    if 'http_application_routing' in addons:
+        addon_profiles['httpApplicationRouting'] = ManagedClusterAddonProfile(enabled=True)
+        addons.remove('http_application_routing')
+    # TODO: can we help the user find a workspace resource ID?
```
[the remark this answers is not in the document]
Comment (mboersma) [reply src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py]: We are already tracking this improvement in VSTS and plan to have it implemented for the subsequent CLI release. I can remove the `# TDOO:` if you prefer but there are several others in this file too.

**Comment by mboersma** [comment]
@troydai the .whl is published to pypi and I removed the temp commit. Let me know what other changes you would like to see.

**Comment by troydai** [comment]
Looks good to me. I'll merge it now.

**Pull request merged**